core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -1340,11 +1338,14 @@ private[spark] class BlockManager(
    */
   private def getPeers(forceFetch: Boolean): Seq[BlockManagerId] = {
     peerFetchLock.synchronized {
-      val cachedPeersTtl = conf.get(config.STORAGE_CACHED_PEERS_TTL) // milliseconds
-      val timeout = System.currentTimeMillis - lastPeerFetchTime > cachedPeersTtl
+      def timeout: Boolean = {
```

Comment:
This changed to a `def` - I think it was fine as a `val`? it's evaluated once, locally.

Reply:
Right, it's called only once but evaluated (in case of `val`) independently from the conditions `cachedPeers == null || forceFetch`. If we would replace `val` by `def`, we could eliminate unnecessary evaluation when `cachedPeers == null` or `forceFetch` are `true`.

Reply:
Yeah, I see it's for lazy evaluation. This isn't that expensive though. It's OK to leave. Another option is to make the "if" condition two nested conditions, and perform this logic inside the first 'if' statement.